src/actions/promote.ts

```diff
@@ -41,4 +50,114 @@ export async function promote(ignoreSslErrors?: boolean) {
+
+    if (!blueGreenDeploymentHelper.isBlueGreenDeploymentStrategy()) {
+        core.debug('Strategy is not blue-green deployment. Invalid request.');
+        throw ('InvalidPromotetActionDeploymentStrategy');
+    }
+
+    let inputManifestFiles: string[] = getManifestFiles(TaskInputParameters.manifests);
+
+    inputManifestFiles = updateContainerImagesInManifestFiles(inputManifestFiles, TaskInputParameters.containers);
+
+    // imagePullSecrets addition
+    inputManifestFiles = updateImagePullSecretsInManifestFiles(inputManifestFiles, TaskInputParameters.imagePullSecrets);
+
+    // checking if rollouts finished and then routing
+    if(blueGreenDeploymentHelper.isIngressRoute()) {
+        let result = await blueGreenDeploymentHelper.blueGreenPromoteIngress(kubectl, inputManifestFiles);
+        const deployedManifestFiles = result.newFilePaths;
+        const resourceTypes: Resource[] = KubernetesObjectUtility.getResources(deployedManifestFiles, models.deploymentTypes.concat([models.DiscoveryAndLoadBalancerResource.service]));
+        await checkManifestStability(kubectl, resourceTypes);
+        blueGreenDeploymentHelper.blueGreenRouteIngress(kubectl, inputManifestFiles, null);
+        blueGreenDeploymentHelper.deleteOppositeIngress(kubectl, inputManifestFiles, blueGreenDeploymentHelper.BLUE_GREEN_NEW_LABEL_VALUE);
+    } else if (blueGreenDeploymentHelper.isSMIRoute()) {
+        let result = await blueGreenDeploymentHelper.blueGreenPromoteSMI(kubectl, inputManifestFiles);
+        const deployedManifestFiles = result.newFilePaths;
+        const resourceTypes: Resource[] = KubernetesObjectUtility.getResources(deployedManifestFiles, models.deploymentTypes.concat([models.DiscoveryAndLoadBalancerResource.service]));
+        await checkManifestStability(kubectl, resourceTypes);
+        blueGreenDeploymentHelper.blueGreenRouteTraffic(kubectl, inputManifestFiles, blueGreenDeploymentHelper.NONE_LABEL_VALUE);
+        blueGreenDeploymentHelper.deleteOpposite(kubectl, inputManifestFiles, blueGreenDeploymentHelper.BLUE_GREEN_NEW_LABEL_VALUE);
+        blueGreenDeploymentHelper.cleanSetUpSMI(kubectl, inputManifestFiles);    
+    } else {
+        core.debug('deleting old deployment and making new ones');
+        let result = await blueGreenDeploymentHelper.blueGreenPromote(kubectl, inputManifestFiles);
+        const deployedManifestFiles = result.newFilePaths;
+        const resourceTypes: Resource[] = KubernetesObjectUtility.getResources(deployedManifestFiles, models.deploymentTypes.concat([models.DiscoveryAndLoadBalancerResource.service]));
+        await checkManifestStability(kubectl, resourceTypes);
+        core.debug('routing to new deployments');
+        blueGreenDeploymentHelper.blueGreenRouteService(kubectl, inputManifestFiles, blueGreenDeploymentHelper.NONE_LABEL_VALUE);
+        blueGreenDeploymentHelper.deleteOpposite(kubectl, inputManifestFiles, blueGreenDeploymentHelper.BLUE_GREEN_NEW_LABEL_VALUE);

```

Comment:
Consider using a different function name.. `deleteOpposite` doesn't set a proper context

Reply:
im deleting the deployment im not targeting now, ive used it in `blue-green-helper` to delete stable versions. should i change it to deleteColor or any other suggestion?